### PR TITLE
Making `cbor` suitable for inclusion in `stdlib`

### PR DESCRIPTION
**Pull request by raoulstrackx**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,17 +17,24 @@ travis-ci = { repository = "pyfisch/cbor" }
 maintenance = { status = "as-is" }
 
 [dependencies]
-half = "1.2.0"
-serde = { version = "1.0.14", default-features = false }
+half = { git = "https://github.com/raoulstrackx/half-rs.git", branch = "raoul/stdlib", default-features = false } #"1.2.0"
+serde = { git = "https://github.com/raoulstrackx/serde.git", branch = "raoul/stdlib", default-features = false }
```
Comment (raoulstrackx) [inline Cargo.toml]: Needs update when other PRs merged

File: Cargo.toml [hunk 2]
```diff
@@ -17,17 +17,24 @@ travis-ci = { repository = "pyfisch/cbor" }
...
+core = { version = "1.0.0", optional = true, package = "rustc-std-workspace-core" }
+alloc = { version = "1.0.0", optional = true, package = "rustc-std-workspace-alloc" }
+compiler_builtins = { version = "0.1.0", optional = true }
 
 [dev-dependencies]
 serde_derive = { version = "1.0.14", default-features = false }
 
+[patch.crates-io]
+serde = { path = "/home/raoul/Documents/serde/serde" }
```
Comment (raoulstrackx) [inline Cargo.toml]: Remove when serde PR merged

File: Cargo.toml [hunk 3]
```diff
@@ -17,17 +17,24 @@ travis-ci = { repository = "pyfisch/cbor" }
...
+
 [features]
 default = ["std"]
 # Uses `alloc` library and adds support for vector functions with
 # `no_std`.
-alloc = ["serde/alloc"]
+#alloc = ["serde/alloc"]
```
Comment (raoulstrackx) [inline Cargo.toml]: TODO: add this to `alloc` dependency

**Comment by raoulstrackx** [comment]
Unfortunately we need to specify dependencies through their git repository directly instead of using a patch section due to https://github.com/rust-lang/cargo/issues/10031

**Review by Pagten — APPROVED** [review]

**Pull request merged**